Updated family view

illusional opened the pull request:
More information, more easily available, with the pedigree. Very early draft

illusional commented:
Show pedigree rows to start
<img width="1158" alt="image" src="https://github.com/populationgenomics/metamist/assets/22381693/37ed0594-92ab-44f8-945e-f7166d7f0d25">

illusional commented:
Family view:

![image](https://github.com/user-attachments/assets/668f1700-a4bf-45e6-8a0c-3329505f31f2)

Participant View:

![image](https://github.com/user-attachments/assets/923dbe2a-bea6-47ab-97c3-62316e49aa7b)

Analysis View: 
![image](https://github.com/user-attachments/assets/2a64729e-aa4f-4201-a1e5-c5e664e324e0)

dancoates commented on web/src/pages/family/FamilyView.tsx:
```diff
@@ -1,229 +1,359 @@
+        }
+    } = {}
+
+    for (const fp of data?.family?.familyParticipants) {
```
This probably needs to be coalesced to something iterable, at the moment if data or family don't exist this'll error out

dancoates commented on web/src/pages/family/FamilyView.tsx:
```diff
@@ -1,229 +1,359 @@
-import iconStyle from '../../shared/iconStyle'
-import SeqPanel from '../../shared/components/SeqPanel'
-import SampleInfo from '../../shared/components/SampleInfo'
+import _ from 'lodash'
```
would be good to import the individual tools here

dancoates commented on web/src/pages/participant/ParticipantViewContainer.tsx:
```diff
@@ -0,0 +1,114 @@
+    }
+
+    const participant = data?.participant
+    const analyses = participant.samples.flatMap((s) =>
```
This is a little thing, but it's a bit more readable to write these nested flatmaps as:

```ts
participant.samples
    .flatMap(s => s.sequencingGroups)
    .flatMap(sg => sg.analyses)
    .flatMap(an => ({
        ...an,
        sgs: an.sequencingGroups.map(sg => sg.id)
    }))
```

rather than calling flatMap in the predicate of each function

dancoates commented on web/src/index.css:
```diff
@@ -490,3 +498,7 @@ html[data-theme='dark-mode'] .ui.table {
 }
+
+.dimmed.dimmable > .ui.modals.dimmer.visible {
+    display: flex !important;
```
Not sure I approve of this haha, but also not too fussed as it seems like we've been fairly liberal with !important in the past

dancoates commented on web/src/shared/components/pedigree/TangledTree.tsx:
```diff
@@ -1,29 +1,774 @@
 /* Inspired by https://observablehq.com/@nitaku/tangled-tree-visualization-ii */
```
I haven't reviewed this file in too much detail but happy for it to go in as-is. My feeling is that it could probably be simplified by moving away from SVG and relying on the DOM layout engine to organize the nodes but I also suspect I'm underestimating the complexity. Will have a bit of a better look at it at a later date.

dancoates reviewed (commented):
Pretty happy with this generally, have made a few very minor comments, but generally happy to approve. Will have another look once you've made the changes you wanted.

illusional commented:
Thanks @dancoates, I've:

- applied your suggestions
- added a quick route for an analysis page, 
- updated a few places to use a link instead of an anchor. Seems to work okay from my local testing!
- Used word-breaks to force a wrapping to make the views appear better (including replacing `<pre>` with a `<span style="font-family: monospace">`)
